src/main/kotlin/app/ui/AddRepoState.kt

```diff
@@ -36,22 +38,24 @@ class AddRepoState constructor(private val context: Context,
                     break  // User finished to add repos.
                 }
             } else {
+                val recursive = UiHelper.confirm("Do you want " +
```

Comment:
I'm not sure if I follow.
If you want to search recursively, you will answer Y then will walk through each subdirectory and validate if it's a valid repository.
Also, this is asked only once, the first time you go through the wizard. 
If I don't ask the question, the directory in question will not be a valid repository since the current `RepoHelper.isValidRepo` validates the current directory.
Do we want to change the `isValidRepo` behaviour?

Reply:
In case the user types a repository path, he still gets the message about whether he wants search recursively. This is weird, no?

Reply:
No, it's not weird.
I still don't follow. What would be the expected behaviour?
When given a path, search it recursively by default?

Reply:
I mean the contrary. With your approach it searches recursively by default in interactive mode even when the user adds repository directly, not a folder with repositories.

Reply:
In that case, should I remove in interactive mode to ask the user if he wants to search recursively and only via the `sourcerer add` would the user need to specify the -r to search recursively?
The current behaviour is that if you go in the interactive mode, you add your repository and it will ask you if you want to search recursively. If you answer `N`, it will just look in the current directory.
You gotta help me out a bit, as I don't understand how it should behave 😞

Reply:
It is better to have the same behavior in each mode.
As I mentioned before it can be solved by checking whether the typed path is a valid repo. In that case, it shouldn't go recursively.

Reply:
Okay, so I should validate from the get-go if the repo is a valid path, if it is a valid one, just do the current behaviour, if it's not, search recursively?

Reply:
Yep.

Reply:
Makes sense. I'll make the changes. 
And sorry for the back and forth 🙃

Reply:
No problem:) Thank you!

Reply:
changed accordingly to the previous comments